telluric/georaster.py

```diff
@@ -1628,6 +1623,36 @@ def get_tile(self, x_tile, y_tile, zoom,
         affine = affine * Affine.scale(ratio, ratio)
         return self.get_window(window, bands=bands, xsize=256, ysize=256, masked=masked, affine=affine)
 
+    def get_tile(self, x_tile, y_tile, zoom,
+                 bands=None, masked=False, resampling=Resampling.cubic):
+        """Convert mercator tile to raster window.
+
+        :param x_tile: x coordinate of tile
+        :param y_tile: y coordinate of tile
+        :param zoom: zoom level
+        :param bands: list of indices of requested bands, default None which returns all bands
+        :return: GeoRaster2 of tile in WEB_MERCATOR_CRS
+
+        You can use TELLURIC_GET_TILE_BUFFER env variable to control the number of pixels surrounding
+        the vector you should fetch when using this method on a raster that is not in WEB_MERCATOR_CRS
+        default to 10
+        """
+        if self.crs == WEB_MERCATOR_CRS:
+            return self._get_tile_when_web_mercator_crs(x_tile, y_tile, zoom, bands, masked, resampling)
+
+        roi = GeoVector.from_xyz(x_tile, y_tile, zoom)
+        left, bottom, right, top = roi.get_bounds(WEB_MERCATOR_CRS)
+        new_affine = rasterio.warp.calculate_default_transform(WEB_MERCATOR_CRS, self.crs,
+                                                               256, 256, left, bottom, right, top)[0]
+        new_resolution = new_affine[0]
+        buffer_ratio = int(os.environ.get("TELLURIC_GET_TILE_BUFFER", 1))
+        roi_buffer = roi.buffer(math.sqrt(roi.area*buffer_ratio/100))
+        raster = self.crop(roi_buffer, resolution=new_resolution, masked=masked, bands=bands)
+        raster = raster.reproject(dst_crs=WEB_MERCATOR_CRS, resolution=MERCATOR_RESOLUTION_MAPPING[zoom],
+                                  dst_bounds=roi_buffer.get_bounds(WEB_MERCATOR_CRS))
+        raster = raster.crop(roi).reproject(dimensions=(256, 256))
```

Comment:
you can use `raster = raster._crop(roi, 256, 256)` and save the reprojection

Reply:
but this is not the same, it is not resampling